### PR TITLE
added connect to app

### PR DESCRIPTION
**Pull request by sonyco-4u**
Hi Reviewer!

In this project, you will connect your React-Redux application to an existing API to create and remove books in a remote server.

I used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls.
I fetched the book list from Bookstore API on the Books page load.
I refactored add book and remove book features to persist your changes in the server.
I made changes by using API.

### CODE REVIEW

**Review by ElsonOtake — CHANGES_REQUESTED** [review]
STATUS: CHANGES REQUIRED 🟡 

Hi @sonyco-4u,

Good job so far! 🎉 🎉 
There are some issues that you still need to work on to go to the next project but you are almost there! 🥲 🥲 

woohoo-youre-almost

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Required Changes ♻️
_Check the comments under the review._

- [ ] Please use the [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls as required in this project. This will accepts a Redux action type and a callback function and return a promise. Your code will be more professional and powerful.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sonyco-4u** [comment]
@ElsonOtake I have made the necessary corrections by making use react toolkit in creating my api's to CreateAsyncThunk.

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,54 +1,42 @@
...
-    case GET_BOOKS_API:
-      return [...action.payload];
+    case ADD:
+      return [...state, action.book];
+
+    case GET:
+      return action.books;
+
+    case REMOVE:
+      return [...state].filter((book) => book.id !== action.id);
```
Comment (ha-manel) [inline src/redux/books/books.js]: - [ ] As requested by the previous reviewer and as mentioned in the project requirements kindly use `createAsyncThunk`  in the project for creating thunk action creators 👍

**Review by ha-manel — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️

## Hi @sonyco-4u 👋 
Great job so far, There are some changes that need to be implemented in order to proceed to the next project 👍  



## Highlights:  
* All tests passed ✔️
* Professional README file ✔️

## Required Changes:
_Kindly check the comments below_

## Optional:
- [ ] Kindly use better structured commit messages, they should be capitalized sentences in the imperative form (ex: "Add linters" instead of "added linters") 👍  

------
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sonyco-4u** [comment]
@ha-manel  I have made the corrections, I now have good commit messages and also I used   createAsyncThunk in the project for creating thunk action creators. Thanks all for your review, am submitting for my last review, so that I can submit this Project.

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @sonyco-4u   

Your project is complete! Great job for making the requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Api calls ok
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**